src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+
+                SSRB(*tmp_projdata_2d_sptr,
+                     *inv_projdata_3d_sptr,false);
+
+                // Crucial: Avoid divisions by zero!!
+                // This should be resolved after https://github.com/UCL/STIR/issues/348
+                pow_times_add min_threshold (0.0f, 1.0f, 1.0f,  1E-20f, NumericInfo<float>().max_value());
+                pow_times_add invert (0.0f, 1.0f, -1.0f, NumericInfo<float>().min_value(), NumericInfo<float>().max_value());
+                apply_to_proj_data(*tmp_projdata_2d_sptr, min_threshold);
+                apply_to_proj_data(*tmp_projdata_2d_sptr, invert);
+
+                norm_coeff_2d_sptr.reset(new BinNormalisationFromProjData(tmp_projdata_2d_sptr));
+                norm_coeff_2d_sptr->set_up(this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone());
+            }
+            else
+            {
+                norm_coeff_2d_sptr.reset(new BinNormalisationFromProjData(tmp_norm_projdata_sptr));
+                norm_coeff_2d_sptr->set_up(this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone());
+            }
+
+            shared_ptr<BinNormalisationFromProjData>atten_coeff_2d_sptr(new BinNormalisationFromProjData(atten_projdata_2d_sptr));
+            this->multiplicative_binnorm_2d_sptr.reset(
+                        new ChainedBinNormalisation(norm_coeff_2d_sptr, atten_coeff_2d_sptr));
+            this->multiplicative_binnorm_2d_sptr->set_up(this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone());
+        }
+
+    }
+#else
+    else
+    {
+        shared_ptr<BinNormalisationFromProjData>atten_coeff_2d_sptr(new BinNormalisationFromProjData(atten_projdata_2d_sptr));
+
+        std::string in_filename = extras_path.get_path() + "tmp_projdata_2d.hs";
+        shared_ptr<ProjData>tmp_projdata_2d_sptr = ProjData::read_from_file(in_filename);
+        shared_ptr<BinNormalisationFromProjData>norm_coeff_2d_sptr(new BinNormalisationFromProjData(tmp_projdata_2d_sptr));
+
+        this->multiplicative_binnorm_2d_sptr.reset(
+                    new ChainedBinNormalisation(norm_coeff_2d_sptr, atten_coeff_2d_sptr));
+        this->multiplicative_binnorm_2d_sptr->set_up(this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone());
+    }
+#endif
+    if (run_in_2d_projdata)
+        iterative_object->get_objective_function_sptr()->set_normalisation_sptr(multiplicative_binnorm_2d_sptr);
+    else
+        iterative_object->get_objective_function_sptr()->set_normalisation_sptr(multiplicative_binnorm_sptr);
+
+    info("ScatterEstimation: Done on normalisation coefficients.");
+
+    //
+    // Set backgroud (randoms) projdata
+    //
+
+    if (!is_null_ptr(this->back_projdata_sptr))
+    {
+#if 0
+        if( back_projdata_sptr->get_num_segments() > 1)
+        {
+            info("ScatterEstimation: Running SSRB on the background data ...");
+
+            FilePath tmp(this->back_projdata_filename);
```

Comment:
in the future, filename might not be set. Let's mark it with a `TODO` comment

Reply:
ok